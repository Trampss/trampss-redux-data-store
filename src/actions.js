const scope = action => `@trampss/${action.toUpperCase()}`

export const SET = prefix => scope(`SET_${prefix}`)
export const set = prefix => payload => ({ type: SET(prefix), payload })

export const RESET = prefix => scope(`RESET_${prefix}`)
export const reset = prefix => () => ({ type: RESET(prefix) })

export const ADD = prefix => scope(`ADD_${prefix}`)
export const add = prefix => payload => ({ type: ADD(prefix), payload })

export const UPDATE = prefix => scope(`UPDATE_${prefix}`)
export const update = prefix => payload => ({ type: UPDATE(prefix), payload })

export const REMOVE = prefix => scope(`REMOVE_${prefix}`)
export const remove = prefix => key => ({ type: REMOVE(prefix), payload: key })

export const ADD_OR_UPDATE = prefix => scope(`ADD_OR_UPDATE_${prefix}`)
export const addOrUpdate = prefix => payload => ({ type: ADD_OR_UPDATE(prefix), payload })
