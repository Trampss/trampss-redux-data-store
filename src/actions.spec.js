/* eslint-env jest */
import { set, remove, add, reset, addOrUpdate } from './actions'

const prefix = 'testPrefix'
const Element = code => ({ code, an: `element_${code}` })

describe('actions', () => {
  it('should trigger a set action', () => expect(set(prefix)([Element('set1'), Element('set2')])).toMatchSnapshot())
  it('should trigger a remove action', () => expect(remove(prefix)('remove')).toMatchSnapshot())
  it('should trigger an add action', () => expect(add(prefix)(Element('add'))).toMatchSnapshot())
  it('should trigger a reset action', () => expect(reset(prefix)()).toMatchSnapshot())
  it('should trigger an addOrUpdate action', () => expect(addOrUpdate(prefix)(Element('add'))).toMatchSnapshot())
})
