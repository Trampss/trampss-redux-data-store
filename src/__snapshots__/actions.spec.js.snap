// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should trigger a del action 1`] = `
Object {
  "payload": "del",
  "type": "@trampss/DEL_TESTPREFIX",
}
`;

exports[`actions should trigger a reset action 1`] = `
Object {
  "type": "@trampss/RESET_TESTPREFIX",
}
`;

exports[`actions should trigger a set action 1`] = `
Object {
  "payload": Array [
    Object {
      "an": "element_set1",
      "code": "set1",
    },
    Object {
      "an": "element_set2",
      "code": "set2",
    },
  ],
  "type": "@trampss/SET_TESTPREFIX",
}
`;

exports[`actions should trigger an add action 1`] = `
Object {
  "payload": Object {
    "an": "element_add",
    "code": "add",
  },
  "type": "@trampss/ADD_TESTPREFIX",
}
`;
